pkg/resources/secret_with_oauth_authorization_code_grant.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var secretAuthorizationCodeGrantSchema = func() map[string]*schema.Schema {
+	secretAuthorizationCodeGrant := map[string]*schema.Schema{
+		"oauth_refresh_token": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the token as a string that is used to obtain a new access token from the OAuth authorization server when the access token expires.",
+		},
+		"oauth_refresh_token_expiry_time": {
+			Type:             schema.TypeString,
+			Required:         true,
+			DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShow("oauth_refresh_token_expiry_time"),
+			Description:      "Specifies the timestamp as a string when the OAuth refresh token expires. Accepted string formats: YYYY-MM-DD, YYYY-MM-DD HH:MI, YYYY-MM-DD HH:MI:SS, YYYY-MM-DD HH:MI <timezone>",
+		},
+		"api_authentication": {
+			Type:             schema.TypeString,
+			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+			Required:         true,
+			Description:      "Specifies the name value of the Snowflake security integration that connects Snowflake to an external service when setting Type to OAUTH2.",
+		},
+	}
+	return helpers.MergeMaps(secretCommonSchema, secretAuthorizationCodeGrant)
+}()
+
+func SecretWithAuthorizationCodeGrant() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateContextSecretWithAuthorizationCodeGrant,
+		ReadContext:   ReadContextSecretWithAuthorizationCodeGrant,
+		UpdateContext: UpdateContextSecretWithAuthorizationCodeGrant,
+		DeleteContext: DeleteContextSecretWithAuthorizationCodeGrant,
+
+		Schema: secretAuthorizationCodeGrantSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportSecretWithAuthorizationCodeGrant,
+		},
+		Description: "Secret with OAuth authorization code grant where Secrets Type attribute is set to OAUTH2.",
+
+		CustomizeDiff: customdiff.All(
+			ComputedIfAnyAttributeChanged(secretAuthorizationCodeGrantSchema, DescribeOutputAttributeName, "oauth_refresh_token_expiry_time"),
+		),
+	}
+}
+
+func ImportSecretWithAuthorizationCodeGrant(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	logging.DebugLogger.Printf("[DEBUG] Starting secret with authorization code import")
+	client := meta.(*provider.Context).Client
+
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	secretDescription, err := client.Secrets.Describe(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := d.Set("oauth_refresh_token_expiry_time", secretDescription.OauthRefreshTokenExpiryTime.String()); err != nil {
```

Comment:
Why this is not read in the Read operation?

Reply:
In Read operation it is read with `setStateToValuesFromConfig` because we don't want to override users `date / datetime` input value with the snowflake timestamp.
Config value for `oauth_refresh_token_expiry_time` should be left just for config, and the value from snowflake should be held in the `describe_output`. Thats what I've discussed with @sfc-gh-asawicki.

Reply:
Yep, we discussed such a workaround to not handle the differences in format/timezone.

Reply:
Ok, I didn't notice it's timestamp